lib/OpenQA/Scheduler/Model/Jobs.pm

```diff
@@ -132,79 +133,138 @@ sub schedule {
-            my $scheduled_state;
+            log_debug("Sent job(s) '$job_ids_str' to worker '$worker_id'");
+
+            # associate the worker to the job, so the worker can send updates
             try {
-                # We associate now the worker to the job, so the worker can send updates.
-                if ($job->set_assigned_worker($worker)) {
-                    push(@successfully_allocated, {job => $allocated->{job}, worker => $allocated->{worker}});
+                if ($actual_job_count > 1) {
+                    my %worker_assignment = (
+                        state              => ASSIGNED,
+                        t_started          => undef,
+                        assigned_worker_id => $worker_id,
+                    );
+                    $_->update(\%worker_assignment) for @jobs;
+                    $worker->update({job_id => $first_job_id});
+                    # note: The job_id column of the workers table is updated as soon as the worker progresses
+                    #       to the next job so the actually current job and current module can be displayed.
                 }
                 else {
-                    # Send abort and reschedule if we fail associating the job to the worker
-                    die "Failed rollback of job" unless $job->reschedule_rollback($worker);
+                    if ($jobs[0]->set_assigned_worker($worker)) {
+                        push(@successfully_allocated, {job => $first_job_id, worker => $worker_id});
+                    }
+                    else {
+                        # Send abort and reschedule if we fail associating the job to the worker
+                        die "Failed rollback of job" unless $jobs[0]->reschedule_rollback($worker);
+                    }
                 }
             }
             catch {
-                log_debug("Failed to set worker in scheduling state :( bummer! Reason: $_");
+                log_debug("Failed to set worker in scheduling state, reason: $_");
             };
-
         }
         else {
-            log_debug("Failed sending job '" . $allocated->{job} . "' to worker '" . $allocated->{worker});
-
+            # reset worker and jobs on failure
+            log_debug("Failed sending job(s) '$job_ids_str' to worker '$worker_id'");
             try {
                 $worker->unprepare_for_work;
             }
             catch {
-                log_debug("Failed resetting unprepare worker :( bummer! Reason: $_");
+                log_debug("Failed resetting unprepare worker, reason: $_");
             };
-
-            try {
-                # Remove the associated worker and be sure to be in scheduled state.
-                die "Failed reset" unless $job->reschedule_state;
+            for my $job (@jobs) {
+                try {
+                    # Remove the associated worker and be sure to be in scheduled state.
+                    die "failed reset" unless $job->reschedule_state;
```

Comment:
That code reads a bit awkward to me because you can't tell if it's a variable or a method call. Maybe worth considering to move the error logging into the method itself?

Not a blocker, just an observation from reading code I'm not super familiar with.

Reply:
That's true. I won't touch it as part of this PR because it was like this before. Adding a refactoring commit on top of the already grown PR is likely a bad idea.